plugins/json-adapter/src/main/java/org/polypheny/db/adapter/json/JsonSource.java

```diff
@@ -0,0 +1,292 @@
+                    yield new URL( files );
+                } catch ( MalformedURLException e ) {
+                    throw new GenericRuntimeException( e );
+                }
+            }
+        };
+    }
+
+
+    @Override
+    public void updateNamespace( String name, long id ) {
+        namespace = new JsonNamespace( name, id, adapterId );
+    }
+
+
+    @Override
+    public Namespace getCurrentNamespace() {
+        return namespace;
+    }
+
+
+    @Override
+    public void shutdown() {
+        removeInformationPage();
+    }
+
+
+    @Override
+    public List<ExportedDocument> getExportedCollections() {
+        try {
+            return JsonMetaRetriever.getDocuments( jsonFiles );
+        } catch ( IOException e ) {
+            throw new GenericRuntimeException( "Failed to retrieve documents from json file." );
+        }
+    }
+
+
+    @Override
+    public AdapterCatalog getCatalog() {
+        return adapterCatalog;
+    }
+
+
+    @Override
+    public void restoreCollection( AllocationCollection allocation, List<PhysicalEntity> entities, Context context ) {
+        PhysicalEntity collection = entities.get( 0 ); // TODO: set breakpoint and take a look at what's in this list...
+        updateNamespace( collection.getNamespaceName(), collection.getNamespaceId() );
+        try {
+            PhysicalCollection physicalCollection = new JsonCollection( JsonMetaRetriever.findDocumentUrl( jsonFiles, collection.getName() ), collection, allocation.getId(), namespace, this );
+            adapterCatalog.addPhysical( allocation, physicalCollection );
+        } catch ( MalformedURLException | NoSuchFileException e ) {
+            throw new GenericRuntimeException( e );
+        }
+    }
+
+
+    @Override
+    public List<PhysicalEntity> createCollection( Context context, LogicalCollection logical, AllocationCollection allocation ) {
+        PhysicalCollection collection = adapterCatalog.createCollection(
+                logical.getNamespaceName(),
+                logical.getName(),
+                logical,
+                allocation
+        );
+        try {
+            PhysicalCollection physicalCollection = new JsonCollection( JsonMetaRetriever.findDocumentUrl( jsonFiles, collection.getName() ), collection, allocation.getId(), namespace, this );
+            adapterCatalog.replacePhysical( physicalCollection );
+            return List.of( physicalCollection );
+        } catch ( MalformedURLException | NoSuchFileException e ) {
+            throw new GenericRuntimeException( e );
+        }
+    }
+
+
+    @Override
+    public void dropCollection( Context context, AllocationCollection allocation ) {
+        // TODO: What is this supposed to do?
+    }
+
+
+    @Override
+    public void renameLogicalColumn( long id, String newColumnName ) {
+        // TODO: Ask David: Why is this part of this interface?
+        log.debug( "NOT SUPPORTED: JSON source does not support method renameLogicalColumn()" );
+    }
+
```

Comment:
Physical stores normally use physical names which are mapped to the logical names before pushing operations to these stores. But depending on the operation this might not be possible. While stores could request this information during push-down it is much simpler and efficient to just keep track of this information for relevant structures in the store as well.

Reply:
Yes